### PR TITLE
Refactor TargetIdName, DistributionSetIdName, SoftwareModuleIdName entities

### PR DESCRIPTION
**Pull request by SirWayne**
Unfortunately I cannot delete the *IdName entities, because of the assignment and deletion action. The assignment and deletion action needs some information for the popup e.g. name, version, controller id.
To avoid unnecessary select statements to get all informations, this entities are reused.

What I have done:

- move TargetIdName, DistributionSetIdName, SoftwareModuleIdName from the repository api to a ui entity package.
- now all tables use the entity#getId as vaadin row id. Remove all *IdName as table row identifier. All LazyQueryContainer use the entity id.
adapt all drag and drop components
- remove expensive getEntity() calls from remote event. There is only 1 table left, which we have to double check (DistributionTable#onDistributionSetUpdateEvents) the calls.
- Fix some NPE's if a entity is deleted by another user

Reviewers: @kaizimmerm  @Jonkno

### CODE REVIEW

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetManagement.java [hunk 1]
```diff
@@ -643,4 +586,27 @@ Long countAllTargetsByTargetFilterQueryAndNotInRolloutGroups(List<RolloutGroup>
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_UPDATE_TARGET + SpringEvalExpressions.HAS_AUTH_OR
             + SpringEvalExpressions.IS_CONTROLLER)
     Target updateTarget(TargetUpdate update);
+
+    /**
+     * Find {@link Target} based on given ID returns found Target without
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetManagement.java]: Please rework the javadoc for a clearer description what the method does. 

e.g.: Find a {@link Target} based a given ID. The returned target will not contain details (e.g {@link Target#getTags()} and {@link Target#getActions()})

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DistributionSetManagementTest.java [hunk 2]
```diff
@@ -750,4 +750,23 @@ public void verifyDistributionSetAssignmentResultNotContainsAlreadyAssignedTarge
         assertThat(distributionSetRepository.findAll()).hasSize(1);
     }
 
+    @Test
+    @Description("Verify that the find all by ids contains the entities which are looking for")
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DistributionSetManagementTest.java]: Add the  @ExpectEvents(...) annotations

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java [hunk 3]
```diff
@@ -721,4 +707,23 @@ public void targetCanBeReadWithOnlyReadTargetPermission() throws Exception {
         });
 
     }
+
+    @Test
+    @Description("Verify that the find all by ids contains the entities which are looking for")
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java]: Add the @ExpectEvents(...) annotations

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java [hunk 4]
```diff
@@ -721,4 +707,23 @@ public void targetCanBeReadWithOnlyReadTargetPermission() throws Exception {
         });
 
     }
+
+    @Test
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java]: I would write: "Verify that the find all **targets** by ids **method** contains the entities **that we** are looking for"

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java [hunk 5]
```diff
@@ -721,4 +707,23 @@ public void targetCanBeReadWithOnlyReadTargetPermission() throws Exception {
...
+        searchIds.add(testdataFactory.createTarget("target-4").getId());
+        searchIds.add(testdataFactory.createTarget("target-5").getId());
+        searchIds.add(testdataFactory.createTarget("target-6").getId());
+        for (int i = 0; i < 9; i++) {
+            testdataFactory.createTarget("test" + i);
+        }
+
+        final List<Target> foundDs = targetManagement.findTargetAllById(searchIds);
+
+        assertThat(foundDs).hasSize(3);
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java]: Add as("...")

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java [hunk 6]
```diff
@@ -721,4 +707,23 @@ public void targetCanBeReadWithOnlyReadTargetPermission() throws Exception {
...
+
+        final List<Long> collect = foundDs.stream().map(Target::getId).collect(Collectors.toList());
+        assertThat(collect).containsAll(searchIds);
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java]: Add as("...")

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DistributionSetManagementTest.java [hunk 7]
```diff
@@ -750,4 +750,23 @@ public void verifyDistributionSetAssignmentResultNotContainsAlreadyAssignedTarge
...
+        searchIds.add(testdataFactory.createDistributionSet("ds-4").getId());
+        searchIds.add(testdataFactory.createDistributionSet("ds-5").getId());
+        searchIds.add(testdataFactory.createDistributionSet("ds-6").getId());
+        for (int i = 0; i < 9; i++) {
+            testdataFactory.createDistributionSet("test" + i);
+        }
+
+        final List<DistributionSet> foundDs = distributionSetManagement.findDistributionSetAllById(searchIds);
+
+        assertThat(foundDs).hasSize(3);
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DistributionSetManagementTest.java]: Add as("...")

File: hawkbit-ui/src/main/resources/messages.properties [hunk 8]
```diff
@@ -549,3 +549,11 @@ controller.polling.title=Polling Configuration
 controller.polling.time=Polling Time
 controller.polling.overduetime=Polling Overdue Time
 
+# entity not exist
+target.not.exists=Target {0} not exists. Maybe the target is deleted
+targets.not.exists=Targets not exists. Maybe the targets are deleted
+
+distributionset.not.exists=Distribution set not exists. Maybe the set is deleted
+distributionsets.not.exists=Distribution sets not exists. Maybe the sets are deleted
```
Comment (Jonkno) [inline hawkbit-ui/src/main/resources/messages.properties]: If the warning is related to multiple Distribution sets better write: Distribution sets do not exists. Maybe the sets were deleted.

File: hawkbit-ui/src/main/resources/messages.properties [hunk 9]
```diff
@@ -549,3 +549,11 @@ controller.polling.title=Polling Configuration
 controller.polling.time=Polling Time
 controller.polling.overduetime=Polling Overdue Time
 
+# entity not exist
+target.not.exists=Target {0} not exists. Maybe the target is deleted
+targets.not.exists=Targets not exists. Maybe the targets are deleted
+
+distributionset.not.exists=Distribution set not exists. Maybe the set is deleted
```
Comment (Jonkno) [inline hawkbit-ui/src/main/resources/messages.properties]: better write: Distribution set does not exist. Maybe the set was deleted.

File: hawkbit-ui/src/main/resources/messages.properties [hunk 10]
```diff
@@ -549,3 +549,11 @@ controller.polling.title=Polling Configuration
 controller.polling.time=Polling Time
 controller.polling.overduetime=Polling Overdue Time
 
+# entity not exist
+target.not.exists=Target {0} not exists. Maybe the target is deleted
```
Comment (Jonkno) [inline hawkbit-ui/src/main/resources/messages.properties]: better write: Target {0} does not exist. Maybe the target was deleted.

File: hawkbit-ui/src/main/resources/messages.properties [hunk 11]
```diff
@@ -549,3 +549,11 @@ controller.polling.title=Polling Configuration
...
+targets.not.exists=Targets not exists. Maybe the targets are deleted
```
Comment (Jonkno) [inline hawkbit-ui/src/main/resources/messages.properties]: better write: Targets do not exists. Maybe the targets were deleted.

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java (hunk 5 above)
Comment (SirWayne) [inline hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/TargetManagementTest.java]: Why should i add `às("...")`? It's not necessary

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetManagement.java (hunk 1 above)
Comment (SirWayne) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetManagement.java]: See `findTargetByControllerID` is the same

**Pull request merged**